### PR TITLE
#10225: Enhance UX for simple 3D navigation in touch devices [Implementation via Config-level]

### PR DESCRIPTION
**Pull request by mahmoudadel54**
## Description
In this PR, the required enhancements are implemented but within the config-level by adding some cfg for cesium mapOptions to enhance UX of navigation.
In includes: 
- Handle disable tilt effect
- Handle enabling zoom limits max, min or both
- Handle enabling fixing orientation of the map scene
- Handle enabling hiding the compass UI widget [will disable tilt effect as well]
- Handling quick reset to the initial camera scene stored into map object

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue
#10225 
**What is the current behavior?**
#10225 

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
There are 2 notes need to highlighted: 
1- For the point `Quick reset of the 3D map view (the camera view) to restore the initial state`, I have checked the implementation of zoomToExtent button and I found out as an enhancement for 3D mode only, we can pass the viewerOptions object - that includes orientation and destination objects - to the zoomTo action which will do the same required functionality.

But If the required logic is keeping zoomTo extent implementation as it is currently and extending the functionality by adding the new functionality in a separate place, it is Ok. 

2- For the point of `fix camera orientation limits and/or fixed camera orientation`, When I enable fix camera orientation ` enableFixedOrientation: true` and `disableTilt = true`, the tilt doesn't disable.
It enables the tilt effect by default. so there will be conflict between disabling tilt and fix camera orientation.

### CODE REVIEW

**Comment by mahmoudadel54** [comment]
Hi 
These are some simple demos demonstrating the UI implementation. I add the configuration into localConfig.json file into defaultMapOptions object --> cesium --> 
image

1- configure `enableFixedOrientation: true` 

https://github.com/geosolutions-it/MapStore2/assets/58145645/2e13f7be-da81-4592-9ee0-82d22ad93388


2- configure `disableTilt: true`


3- configure `hideCompass: true` 

https://github.com/geosolutions-it/MapStore2/assets/58145645/832d9dc7-41b3-44c1-9423-34ffff5c99d1

4- configure `enableQuickResetTo3DView: true`


https://github.com/geosolutions-it/MapStore2/assets/58145645/d4d07ab4-4519-4845-aaf4-a558f37e59e4

5- configure `zoomLimits: {
"min": 20000,
 "max": 20000000
}`

https://github.com/geosolutions-it/MapStore2/assets/58145645/426a868c-dff3-4b84-a245-ad21ef5ed435

File: web/client/components/map/cesium/Map.jsx [hunk 1]
```diff
@@ -151,15 +152,31 @@ class CesiumMap extends React.Component {
...
+                // sometimes the center is undefined on browser history navigation
+                // we should not perform setView in these cases
+                if (initDestination[0] !== undefined && initDestination[1] !== undefined) {
+                    map.camera.setView({
+                        destination: Cesium.Cartesian3.fromDegrees(...initDestination),
+                        orientation: this.props?.mapInitialConfig?.viewerOptions?.orientation
+                    });
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (mahmoudadel54) [reply web/client/components/map/cesium/Map.jsx:171]: @allyoucanmap I noticed that and I have mentioned similar into the issue description at the `Other useful information` section

File: web/client/components/map/cesium/Map.jsx [hunk 2]
```diff
@@ -135,7 +136,7 @@ class CesiumMap extends React.Component {
         if (this.props.registerHooks) {
             this.registerHooks();
         }
-        if (this.props.mapOptions?.navigationTools !== false) {
+        if (this.props.mapOptions?.navigationTools !== false && !this.props.mapOptions?.hideCompass) {
```
[the remark this answers is not in the document]
Comment (mahmoudadel54) [reply web/client/components/map/cesium/Map.jsx:139]: @allyoucanmap 
I thought the `navigationTools` is for hide the compass but with keeping rotation/zoom/tilt actions working so hideCompass will hide with prevent rotation/zoom/tilt actions and allowing just pan action.
@tdipisa please, correct me if it is not

File: web/client/components/map/cesium/Map.jsx [hunk 3]
```diff
@@ -180,6 +197,39 @@ class CesiumMap extends React.Component {
 
         // this is needed to display correctly intersection between terrain and primitives
         scene.globe.depthTestAgainstTerrain = this.props.mapOptions?.depthTestAgainstTerrain ?? false;
+        // handle disable tilt or hide compass ui widget from map
+        if (this.props.mapOptions?.hideCompass || this.props.mapOptions?.disableTilt) {
+            // disable tilt effect if disableTilt into mapOptions = true
+            map.scene.screenSpaceCameraController.enableTilt = false;
+        }
```
[the remark this answers is not in the document]
Comment (mahmoudadel54) [reply web/client/components/map/cesium/Map.jsx:204]: @allyoucanmap 
It is mentioned at the last point in this comment: 
https://github.com/geosolutions-it/MapStore2/issues/10225#issuecomment-2070067927

File: web/client/components/map/cesium/Map.jsx [hunk 4]
```diff
@@ -648,7 +698,7 @@ class CesiumMap extends React.Component {
         this.map.scene.screenSpaceCameraController.enableZoom = !(interactionsOptions.mouseWheelZoom === false);
         this.map.scene.screenSpaceCameraController.enableRotate = !(interactionsOptions.dragPan === false);
         this.map.scene.screenSpaceCameraController.enableTranslate = !(interactionsOptions.dragPan === false);
-        this.map.scene.screenSpaceCameraController.enableTilt = !(interactionsOptions.dragPan === false);
+        this.map.scene.screenSpaceCameraController.enableTilt = !(interactionsOptions.dragPan === false);          // block zoom into background
```
[the remark this answers is not in the document]
Comment (mahmoudadel54) [reply web/client/components/map/cesium/Map.jsx:701]: @allyoucanmap 
I means preventing the user from going underground, I think it need to edited

**Review by allyoucanmap — APPROVED** [review]

**Comment by allyoucanmap** [comment]
@ElenaGallo please test on dev and let us know if we can backport to 2024.01.xx, thanks

**Comment by mahmoudadel54** [comment]
> @ElenaGallo please test on dev and let us know if we can backport to 2024.01.xx, thanks

Hi @ElenaGallo 

The results of this PR you can test [they are just 2 things]:
- maximumZoomDistance, minimumZoomDistance properties can be appended into mapOptions in localConfig and you can see its effect
- zoom to initial view [zoomToExtent button] in the bottom navigation toolbar, you can see the difference in 3D mode

For the rest points mentioned on the issue description, they may be in separate future issues.

**Pull request merged**